### ISSUE TITLE
Bug: rule name prefixed with some characters works in deno-lint-ignore

### ISSUE THREAD

**Issue by magurotuna** [denoland/deno_lint#287]
```ts
// deno-lint-ignore xxxxxxxxxxxxban-types
let a: Object;
```

You can see that the rule name is wrong, prefixed with many `x` characters. We expect this directive not to have an effect.
However, it __does works__, just as we write a directive correctly like `// deno-lint-ignore ban-types`.

On the other hand, rule names __suffixed__ with characters like `ban-typesxxxxxxxx`__do not__ work, which is the behavior we expect.

I read the code and then found the cause at: https://github.com/denoland/deno_lint/blob/14f1543c1684a6a5a6b080bb6695b49e9a8700f5/src/linter.rs#L103-L108

The comment says that using `ends_with` is to skip `@typescript-eslint` prefix.
This means we want `deno-lint` to be able to handle rule names prefixed with `@typescript-eslint`, right?
If so, we could probably realize it by removing `@typescript-eslint` prefix (if exists) when parsing a comment in the function `parse_ignore_comment`. In this way, I guess the bug I mentioned above will be fixed.

**Comment by bartlomieju** [comment]
@magurotuna that's correct. To be honest neither of these solutions is perfect and it should be probably handled by having "aliases" for rules, but I haven't yet figured how to do that. 

It was changed in https://github.com/denoland/deno_lint/pull/269. Feel free to revert to previous solution.

**Comment by magurotuna** [comment]
@bartlomieju hmm... it's kind of more complex problem than I thought. :-|
Could you tell me why you think my solution (removing prefix when parsing) is not perfect? It won't work in some cases?

**Comment by bartlomieju** [comment]
> @bartlomieju hmm... it's kind of more complex problem than I thought. :-|
> Could you tell me why you think my solution (removing prefix when parsing) is not perfect? It won't work in some cases?

It's still hardcoded and won't work for rules with other plugins - but I guess we can not care ATM. So feel free to bring back previous behavior.

**Comment by magurotuna** [comment]
Thank you for the explanation. Understood.
I think this bug is not so serious. So it's OK to pend until some great solution occurs to us.

**Comment by bartlomieju** [comment]
@magurotuna I've been pondering on this issue for some time. I think we can solve it by using `RuleContext` instead of `LintContext` - that way each rule could register multiple "codes" it recognizes (eg. `ban-types`, `@typescript-eslint/ban-types`). 

Ignoring of produced diagnostics would happen when processing each lint rule instead of doing it in bulk at the end. 

That should help us get rid of the hacky code you pointed in the first comment.

**Comment by magurotuna** [comment]
@bartlomieju 
sounds great, that way we'll be able to solve this problem.

**Comment by bartlomieju** [comment]
After some more pondering I think we shouldn't support aliases after all and require the proper rule names from `deno_lint`.

**Comment by bartlomieju** [comment]
@magurotuna what do you think?

**Comment by magurotuna** [comment]
@bartlomieju 

> we shouldn't support aliases after all and require the proper rule names from deno_lint

that makes sense to me.
